### ISSUE TITLE
Corrupted unpacking of Xcode with unxip 1.1.2

### ISSUE THREAD

**Issue by zcutlip** [saagarjha/unxip#23]
When I unpack Xcode 14.1 RC 2, the resulting application bundle is corrupt and won't launch. When I use `/usr/bin/xip` it works fine.

I'm not sure what the issue is, but right off I know `Xcode.app/Contents/Info.plist` is corrupted.

Unpacked with 'xip':
```
$ sha256sum ./Xcode.app/Contents/Info.plist
847d9996dbdbae2916775368c26e3fe3c2a34eb0e738a700368f6f637fb4608c  ./Xcode.app/Contents/Info.plist
```

Unpacked with `unxip` 1.1.2:

```
$ sha256sum /Applications/Xcode.app/Contents/Info.plist
b798bf5b5850661c83a72022a13af0e62d71679b09485b4f95b8e66e2f4e5e25  /Applications/Xcode.app/Contents/Info.plist

$ plutil -convert xml1 -o - Xcode.app/Contents/Info.plist
Xcode.app/Contents/Info.plist: Property List error: Unexpected character b at line 1 / JSON error: JSON text did not start with array or object and option to allow fragments not set. around line 1, column 0.
```

`unxip` hash:
```
$ sha256sum "$(which unxip)"
253b5d2e4602cd52eef1c01d5280802d72c8a55cdc7711ad9a16dd543a84d19c  /opt/homebrew/bin/unxip
```

Xcode hash:
```
$ sha256sum Xcode_14.1_Release_Candidate_2.xip
12f8a3aef78bf354470ad8b351addd925c8edad888137d138ca50a8130eb9f2f  Xcode_14.1_Release_Candidate_2.xip
```

macOS version:
```
$ sw_vers
ProductName:		macOS
ProductVersion:		13.0
BuildVersion:		22A380
```

I've attached the corrupted Info.plist (zipped because GitHub doesn't accept .plist)

Let me know if I can provide anything else

[Info.plist.zip](https://github.com/saagarjha/unxip/files/9912394/Info.plist.zip)

**Comment by saagarjha** [comment]
Hmm, perhaps related to #21. Can you reproduce this? I'll try running it a couple times on my side to see if I can get it to occur.

**Comment by zcutlip** [comment]
I saw #21 and wondered.

This reproduces consistently for me though. I'll give it a try on a different Xcode xip, and see if the problem is just this one.

fwiw, I'm downloading Xcode from developer.apple.com using `aria2c`.

**Comment by zcutlip** [comment]
Just noticed Xcode 14.1 is officially out today, so I downloaded it (same way as before). I `unxip`ed it. It also won't launch, and the Info.plist is also corrupted (other things may be corrupted also, that's just what I've checked).

**Comment by saagarjha** [comment]
The fix for this should be shipping in [1.1.3](https://github.com/saagarjha/unxip/releases/tag/v1.1.3). I added CI to help catch things like this, but let me know if the update doesn't fix things for you.

**Comment by zcutlip** [comment]
It's working for me now. Thanks for the fast turnaround!
